File: JAVA/Myintent/app/src/main/java/com/example/myintent/MainActivity.java
package com.example.myintent;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button =(Button)findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                //메뉴 액티비티를 띄어주는 인텐트
                Intent intent =  new Intent(getApplicationContext(),MenuActivity.class);
                //시스템으로 인텐트를 전달하면서 화면을 띄어준다!
                startActivityForResult(intent,101);
            }
        });
    }

    //응답을 받을 메소드 작성

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if(requestCode ==101){
            String name= data.getStringExtra("name");
            Toast.makeText(getApplicationContext(),"메뉴화면으로부터 응답:"+ name ,Toast.LENGTH_LONG).show();

        }
    }
}

File: Kotlin/NumberLottery/app/src/main/java/com/example/numberlottery/MainActivity.kt
package com.example.numberlottery

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.NumberPicker
import android.widget.TextView
import android.widget.Toast
import androidx.core.view.isVisible

class MainActivity : AppCompatActivity() {

    private val clearButton : Button by lazy{
        findViewById(R.id.btn_clear)
    }

    private val addButton :Button by lazy {
        findViewById(R.id.btn_add)
    }

    private val runButton: Button by lazy {
        findViewById(R.id.btn_run)
    }

    private val numTextViewList : List<TextView> by lazy {
        listOf<TextView>(
                findViewById(R.id.tv_num1)
                ,findViewById(R.id.tv_num2)
                ,findViewById(R.id.tv_num3)
                ,findViewById(R.id.tv_num4)
                ,findViewById(R.id.tv_num5)
                ,findViewById(R.id.tv_num6))
    }

    private var didRun = false
    private val pickNumSet = hashSetOf<Int>()

    private val numPick : NumberPicker by lazy { findViewById(R.id.np_num) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        numPick.minValue = 1
        numPick.maxValue = 45

        initRunButton()
        initAddButton()
        initClearButton()
    }

    private fun initAddButton() {
        addButton.setOnClickListener {
            if(didRun){
                Toast.makeText(this,"초기화 후에 시도해주세요.",Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if(pickNumSet.size >= 5){
                Toast.makeText(this,"숫자는 최대 5개까지 선택할 수 있습니다.",Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if(pickNumSet.contains(numPick.value)){
                Toast.makeText(this,"이미 선택한 숫자입니다..",Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val textView = numTextViewList[pickNumSet.size]
            textView.isVisible = true
            textView.text = numPick.value.toString()

            pickNumSet.add(numPick.value)
        }
    }

    private fun initClearButton(){
        clearButton.setOnClickListener {
            pickNumSet.clear()
            numTextViewList.forEach{
                it.isVisible = false
            }

            didRun = false
        }
    }

    private fun initRunButton() {
        runButton.setOnClickListener {
            val list = getRandom()
            didRun = true

            list.forEachIndexed { index, number ->
                val textView = numTextViewList[index]

                textView.text = number.toString()
                textView.isVisible = true
            }
        }
    }

    //랜덤으로 숫자 반환시켜주기
    private fun getRandom(): List<Int> {
        val numList = mutableListOf<Int>()
                .apply {
                    for(i in 1..45){
                        if(pickNumSet.contains(i)){
                            continue
                        }
                        this.add(i)
                    }
                }
        numList.shuffle()

        val newList = pickNumSet.toList() + numList.subList(0,6- pickNumSet.size)
        return newList.sorted()
    }


}